controller/contour/controller.go

```diff
@@ -149,6 +151,9 @@ func (r *Reconciler) ensureContour(ctx context.Context, contour *operatorv1alpha
 	if err := r.ensureEnvoyService(ctx, contour); err != nil {
 		return fmt.Errorf("failed to ensure service for contour %s/%s: %w", contour.Namespace, contour.Name, err)
 	}
+	if err := r.syncContourStatus(ctx, contour, deploy, ds); err != nil {
```

Comment:
does it matter that if one of the `ensureXXX` functions returns an error above so `ensureContour` returns early, that no status will be created? I wonder if it'd be better/possible to update it as you go, or always update it before returning.